core/types/block_ext_test.go

```diff
@@ -262,22 +203,7 @@ func headerHashComparer() cmp.Option {
 func TestBodyExtraRLP(t *testing.T) {
 	t.Parallel()
 
-	tx := NewTransaction(0, common.Address{1}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
-
-	uncle := &Header{ParentHash: common.Hash{6}}
-	uncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
-	SetHeaderExtra(uncle, uncleExtra)
-
-	body := &Body{
-		Transactions: []*Transaction{tx},
-		Uncles:       []*Header{uncle},
-		Withdrawals:  []*ethtypes.Withdrawal{{Index: 7}}, // ignored
-	}
-	extra := &BlockBodyExtra{
-		Version: 1,
-		ExtData: &[]byte{2},
-	}
-	extras.Body.Set(body, extra)
+	body, wantExtra := bodyWithNonZeroFields()
```

Comment:
I would re-call `bodyWithNonZeroFields()` later to generate a fresh copy of the two objects. It's a simple code change that gives strong guarantees in return.